### PR TITLE
ship 0.29.0 (new formula)

### PR DESCRIPTION
**Pull request by fshot**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by fxcoudert** [comment]
```
==> Downloading https://github.com/replicatedhq/ship/archive/v0.22.0.tar.gz
==> Downloading from https://codeload.github.com/replicatedhq/ship/tar.gz/v0.22.0
Warning: Formula reports different SHA256: 697a8d3186fbce08b95f281b389c6aaf2ad3dfcbc8c5a41506097b47a7f77be7
Downloaded to: /Users/brew/Library/Caches/Homebrew/downloads/b45a9bb7120e78191b5aa77b338bce0abdf0f5b8c3a090c705e25b4866895a66--ship-0.22.0.tar.gz
SHA256: 9c10beea62c0179b3f6b115dcf0bd147bde86b8d17eb5f78e783ab3e5f0252d2
```

**Comment by fshot** [comment]
Thanks, @fxcoudert .  I've confirmed and fixed the SHA256 checksum.

Note, the SHA256 I originally used was the one suggested when I initially ran "brew create" from the ship-0.22.0 tarball.  That same '697a8d' SHA is what's suggested when I try `brew fetch ship --build-from-source`.

```
$ brew fetch ship --build-from-source
==> Downloading https://github.com/replicatedhq/ship/archive/v0.22.0.tar.gz
Already downloaded: /Users/frank/Library/Caches/Homebrew/downloads/b45a9bb7120e78191b5aa77b338bce0abdf0f5b8c3a090c705e25b4866895a66--ship-0.22.0.tar.gz
SHA256: 697a8d3186fbce08b95f281b389c6aaf2ad3dfcbc8c5a41506097b47a7f77be7
Warning: Formula reports different SHA256: 9c10beea62c0179b3f6b115dcf0bd147bde86b8d17eb5f78e783ab3e5f0252d2
```

I'm curious about how this other SHA is being calculated...

**Comment by fshot** [comment]
Hi @fxcoudert, we released a new version of ship this week.  Would you recommend that we wait until this PR is approved & merged before submitting a new PR?  Or edit this PR?

Thanks in advance for your help.

**Comment by fxcoudert** [comment]
@fshot edit this PR please, but squash everything into a single commit

**Comment by fshot** [comment]
Thanks @fxcoudert -- updated & squashed

**Comment by fshot** [comment]
@fxcoudert or other maintainers... any tips on why the brew-bot CI checks haven't completed yet?  Is there a way to review the logs?

**Comment by fshot** [comment]
@BrewTestBot test this please

**Comment by fshot** [comment]
@BrewTestBot test this please

**Comment by fshot** [comment]
@BrewTestBot test this please

(Last test failure was due to an issue with Jenkins build pipeline)

**Comment by fshot** [comment]
@BrewTestBot test this please

File: Formula/ship.rb
Comment (fxcoudert) [inline Formula/ship.rb]: Why does this need to be created manually?

File: Formula/ship.rb
Comment (fxcoudert) [inline Formula/ship.rb]: Why do we need to have `bin` in the `PATH`?

File: Formula/ship.rb
Comment (fxcoudert) [inline Formula/ship.rb]: Non-parallel build is considered a bug and should be reported upstream

File: Formula/ship.rb
Comment (fxcoudert) [inline Formula/ship.rb]: We need a test that exercises the some of the functionality of the app.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/ship.rb
Comment (fshot) [reply Formula/ship.rb]: This was a workaround for an issue that we've been able to resolve within the ship Makefile.  Will remove this from the formula as soon as we release ship 0.27.0 with that fix.

File: Formula/ship.rb
Comment (fshot) [reply Formula/ship.rb]: This was a workaround for an issue that we've been able to resolve within the ship Makefile.  Will remove this from the formula as soon as we release ship 0.27.0 with that fix.

File: Formula/ship.rb
Comment (fshot) [reply Formula/ship.rb]: Hi @fxcoudert ,
I am working on isolating the parallel build problems (which appear to be due to yarn concurrently downloading some of the same packages to the same cache location) so that I can either workaround or report them properly.  Hoping that we can merge the current formula, and then update with a de-deparallelization just as soon as that's resolved.

Cheers,
Frank

File: Formula/ship.rb
Comment (fshot) [reply Formula/ship.rb]: Removed offending path manipulations

File: Formula/ship.rb
Comment (fshot) [reply Formula/ship.rb]: Removed

File: Formula/ship.rb
Comment (fshot) [reply Formula/ship.rb]: Additional tests added.

**Comment by fshot** [comment]
Hi @fxcoudert , would you mind taking another look at this formula?  All your previous comments have been addressed.

Thanks,
Frank

**Comment by igas** [comment]
@BrewTestBot test this please

File: Formula/ship.rb
Comment (igas) [inline Formula/ship.rb]:
```suggestion
    ENV["GOPATH"] = buildpath
    srcpath = buildpath/"src/github.com/replicatedhq/ship"
    srcpath.install buildpath.children
    srcpath.cd do
      system "make", "VERSION=#{version}", "build-minimal"
      bin.install "bin/ship"
    end
```

**Comment by igas** [comment]
```suggestion
  def install
    ENV["GOPATH"] = buildpath
    srcpath = buildpath/"src/github.com/replicatedhq/ship"
    srcpath.install buildpath.children
    srcpath.cd do
      system "make", "VERSION=#{version}", "build-minimal"
      bin.install "bin/ship"
    end
  end
```

**Comment by fshot** [comment]
Thanks, @igas - I like the suggestions.  It's clearer, and works fine locally.  Cheers!

File: Formula/ship.rb
Comment (igas) [inline Formula/ship.rb]: it redirects to https://www.replicated.com/ship

I suggest to use it directly

File: Formula/ship.rb
Comment (igas) [inline Formula/ship.rb]: as mentioned before, this is very primitive checks, can we do `ship init` and check that some files been created/modified?

From the [docs](https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula):

> We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. However, a bad test is better than no test at all.

**Review by igas — DISMISSED** [review]

File: Formula/ship.rb
Comment (fshot) [reply Formula/ship.rb]: Thanks, @igas -- homepage URL is updated

File: Formula/ship.rb
Comment (fshot) [reply Formula/ship.rb]: Thanks, @igas -- I've added a simple test of the base 'ship init' functionality, following the suggestions from the docs (docs referenced examples https://github.com/Homebrew/homebrew-core/blob/master/Formula/tinyxml2.rb and https://github.com/Homebrew/homebrew-core/blob/master/Formula/cmake.rb)

File: Formula/ship.rb
Comment (fxcoudert) [inline Formula/ship.rb]: No doc, no man page?

File: Formula/ship.rb
Comment (fxcoudert) [inline Formula/ship.rb]: Please remove capitals on words "overhead", "maintaining", etc.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/ship.rb
Comment (fshot) [reply Formula/ship.rb]: No, we don't currently have a man page for ship.  Ship docs and description are available at www.replicated.com/ship and the README in the code repo.

File: Formula/ship.rb
Comment (fshot) [reply Formula/ship.rb]: Thanks, these are fixed now.

**Comment by fshot** [comment]
@BrewTestBot test this please

Last build seems to have failed due to transient Github service issues

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @fshot for contributing to Homebrew!

**Comment by fshot** [comment]
Thank you @fxcoudert and @igas -- we are happy to be part of the project!

**Pull request closed without merging**
